lib/src/collections/observable_set.dart

```diff
@@ -116,3 +133,86 @@ class _DelegatingObservableSet<E> extends DelegatingSet<E>
     removeWhere((e) => !test(e));
   }
 }
+
+class _UnmodifiableEmptySet<E> extends IterableBase<E> implements Set<E> {
```

Comment:
is this mainly intended as an optimization? Otherwise could it just use an _UnmodifiableObservableSet of an empty const Set?

the reason I ask ... it might not actually be faster overall. Calling methods on an empty set will be fast, but now the VM has to deal with two subclasses (the empty and non-empty being distinct), which slows down dispatch and prevents inlining if they both hit the same call site. I feel like if the corelibs didn't have a class like this, it probably isn't needed here.